blog/2024-01-23-state-machines-whats-in-a-name/index.mdx

```diff
@@ -0,0 +1,186 @@
+
+<p>
+  <ThemedImage
+    alt="An example of parallel states taken from Statecharts: A visual formalism for complex systems by David Harel."
+    sources={{
+      light:
+        '/blog/2024-01-23-state-machines-whats-in-a-name/orthogonality.png',
+      dark: '/blog/2024-01-23-state-machines-whats-in-a-name/orthogonality.png',
+    }}
+  />
+</p>
+
+Some of the examples later in the paper did demonstrate the use of state and event labels but these were usually single words or short phrases without attention paid to style. Interestingly enough, this example makes use of a **`dot.case`** style for event names which you can [jump ahead an read about here](#xstate-v5-and-dotcase).
+
+<p>
+  <ThemedImage
+    alt="An example of states and events from a statechart depicting the alarm on a Citizen Quartz watch."
+    sources={{
+      light:
+        '/blog/2024-01-23-state-machines-whats-in-a-name/watch-batteries.png',
+      dark: '/blog/2024-01-23-state-machines-whats-in-a-name/watch-batteries.png',
+    }}
+  />
+</p>
+
+By the time the book, "Constructing the User Interface with Statecharts" (Horrocks, 1999) was published, more descriptive state and event labels were present in elaborate examples of statecharts used to model hardware and software systems. This example of a CD player (remember those?) is rather detailed and also quite clear in its use of names and text annotations on the statechart. You can [view this state machine inspired by this example](https://stately.ai/registry/editor/6541bdd9-e250-47c8-9a5e-fa1352cd5088?mode=design&machineId=8ead8108-3959-4db2-a335-9411a19a3416) or [see the full example](https://stately.ai/registry/editor/6541bdd9-e250-47c8-9a5e-fa1352cd5088?mode=design&machineId=1df8a151-0c54-4aac-be55-21560360c311) borrowed from the book.
+
+<p>
+  <ThemedImage
+    alt="Part of a statechart for a physical CD player including states for when the CD is playing, stopped, paused, and events transitioning the machine from state to state."
+    sources={{
+      light: '/blog/2024-01-23-state-machines-whats-in-a-name/horrocks-cd.png',
+      dark: '/blog/2024-01-23-state-machines-whats-in-a-name/horrocks-cd.png',
+    }}
+  />
+</p>
+
+Over the decades since those early writings on statecharts, the emergence of nearly all of the major programming langues in use today have popularized a number of naming conventions for code. Here are some of the most common styles you can choose from when naming states, events, actions, and guards in XState:
```

Comment:
```suggestion
Over the decades since those early writings on statecharts, the emergence of nearly all of the major programming languages in use today have popularized a number of naming conventions for code. Here are some of the most common styles you can choose from when naming states, events, actions, and guards in XState:
```